flink-tests/src/test/java/org/apache/flink/test/state/operator/restore/AbstractOperatorRestoreTestBase.java

```diff
@@ -71,7 +80,17 @@
  * AbstractOperatorRestoreTestBase#migrateJob}, please create the corresponding test resource
  * directory and copy the _metadata file by hand.
  */
-public abstract class AbstractOperatorRestoreTestBase extends TestLogger {
+public abstract class AbstractOperatorRestoreTestBase extends TestLogger implements MigrationTest {
+
+    private static final FlinkVersion MIGRATION_BASE_VERSION = FlinkVersion.v1_16;
```

Comment:
can't we add a utility method `FlinkVersion.previousMinorVersion()` that decrements the minor version by 1? That's how we can derive the base version from the target version instead of having a fixed field which we would have to update again with each release.

Reply:
The `MIGRATION_BASE_VERSION` should only be used in the process to generate the snapshots, thus it looks to me that it might not be necessary to always generate the new snapshots from exactly the previous version, and we do not modify this field for new versions, WDYT?

Reply:
I feel like we need more documentation here, i.e. in-code comments. What about not using a dedicated final field here. We're using it in one place only, anyway. Additionally, can't we use 1.8 to make it clearer that the version doesn't matter?